### DISCUSSION TITLE
Conventional Commits and Auto ReleaseNotes

### DISCUSSION THREAD

**Discussion by CruzanCaramele** [Dynatrace/dynatrace-configuration-as-code#253] [Archive] [answered]
As We intend to generate Release notes automatically using GitHub Actions, I have found an action that would let us do this and produce release notes such as the one in the picture below:

![image](https://user-images.githubusercontent.com/6834860/110914588-dae6a580-8316-11eb-9f94-eb96fb668fb6.png)

However, to go in this direction, all commits have to follow Conventional Commit standard which is found on this page --> https://www.conventionalcommits.org/en/v1.0.0/

I have an action that can check for commit compliance upon a pull request and fail if the standard is not followed, this means we would have to force a contributor to repeat the commit and update the pull request for the compliance check to pass

@tatomir146 @didiladi @patrickpichler @UnseenWizzard   what do you think ?

**Comment by patrickpichler** [comment, marked as answer]
I personally like it to be honest :+1:

**Reply by UnseenWizzard** [reply]
Agreed, I like both using conventional commits and the look of the release note screenshot

What is the tool/action that generates the release notes @CruzanCaramele ?

**Reply by CruzanCaramele** [reply]
@UnseenWizzard  I am using the one from this project --> https://github.com/aevea/release-notary
I tested it with one of my repos, after researching a few, its the one I see that is more maintained

**Comment by UnseenWizzard** [comment]
One more related question @dynatrace-oss/mac-maintainers: 
The conventional commit spec only defines fix & feat as types, but allows for more (see my comment on the open PR on which the release note generator supports visually) 

From a recent look at it, the types outlined by Angular sound like a reasonable base set of types. 
https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type

We should agree what we adopt and note it down in CONTRIBUTING.md! 

Not sure if we want to use scopes?
